addons/stock/report/stock_forecasted.py

```diff
@@ -244,24 +244,26 @@ def _get_out_move_taken_from_stock_data(out, currents, reserved_data):
                 'taken_from_stock': taken_from_stock_out,
             }
 
-        def _reconcile_out_with_ins(lines, out, ins, demand, product_rounding, only_matching_move_dest=True, read=True):
-            index_to_remove = []
-            for index, in_ in enumerate(ins):
-                if float_is_zero(in_['qty'], precision_rounding=product_rounding):
-                    index_to_remove.append(index)
-                    continue
-                if only_matching_move_dest and in_['move_dests'] and out.id not in in_['move_dests']:
+        map_dests_to_ins_per_product, map_in_id_to_data = defaultdict(lambda: defaultdict(OrderedSet)), defaultdict()
```

Comment:
You can use a `dict` for `map_in_id_to_data` as there's not a lot of point having a `defaultdict` without a factory.